src/containers/AdminCampaignEdit.jsx

```diff
@@ -362,43 +363,53 @@ export class AdminCampaignEditBase extends React.Component {
       if (newCampaign.hasOwnProperty("interactionSteps")) {
         newCampaign.interactionSteps = makeTree(newCampaign.interactionSteps);
       }
-
+      
       await this.props.mutations.editCampaign(
         this.props.campaignData.campaign.id,
         newCampaign
-      );
-    }
-  };
-
-  checkSectionSaved(section) {
-    // Tests section's keys of campaignFormValues against props.campaignData
-    // * Determines greyness of section button
-    // * Determine if section is marked done (in green) along with checkSectionCompleted()
-    // * Must be false for a section to save!!
-    // Only Contacts section implements checkSaved()
-    if (section.hasOwnProperty("checkSaved")) {
-      return section.checkSaved();
+        );
+      }
+    };
+    
+    checkSectionSaved(section) {
+      // Tests section's keys of campaignFormValues against props.campaignData
+      // * Determines greyness of section button
+      // * Determine if section is marked done (in green) along with checkSectionCompleted()
+      // * Must be false for a section to save!!
+      // Only Contacts section implements checkSaved()
+      if (section.hasOwnProperty("checkSaved")) {
+        return section.checkSaved();
+      }
+      const sectionState = {};
+      const sectionProps = {};
+      section.keys.forEach(key => {
+        sectionState[key] = this.state.campaignFormValues[key];
+        sectionProps[key] = this.props.campaignData.campaign[key];
+      });
+      if (JSON.stringify(sectionState) !== JSON.stringify(sectionProps)) {
+        return false;
+      }
+      // console.log("SectionState", sectionState)
+      // console.log("SectionProps", sectionProps)
+      return true;
     }
-    const sectionState = {};
-    const sectionProps = {};
-    section.keys.forEach(key => {
-      sectionState[key] = this.state.campaignFormValues[key];
-      sectionProps[key] = this.props.campaignData.campaign[key];
-    });
-    if (JSON.stringify(sectionState) !== JSON.stringify(sectionProps)) {
-      return false;
+    
+    checkSectionCompleted(section) {
+      return section.checkCompleted();
     }
-    return true;
-  }
-
-  checkSectionCompleted(section) {
-    return section.checkCompleted();
-  }
-
-  sections() {
-    const pendingJobs = this.props.campaignData.campaign.pendingJobs;
-    const finalSections = [
-      {
+    
+    sections() {
+      console.log("THIS.PROPS in SECTIONS METHOD in ADMINCAMPAIGNEDIT", this.props);
+      console.log("THIS.STATE in SECTIONS METHOD in ADMINCAMPAIGNEDIT", this.state);
```

Comment:
I spent most of my time scouring through this state and props trying to find the contactData or the extraProps.  When I realized that the props that were located on the other files didn't have half the variables I needed, it was this console.log that was helping me understand where I would find information in the state versus in the props.

The only bummer is that it still doesn't expose a contact, even though I can see the contactsCount perfectly.

I eventually resigned to simply ingesting the filename from the ingestMethod to scrape the first line and feed it to the applyScript method within AdminTextPreview.